### PR TITLE
[fix][broker] Fix redirect loop when using ExtensibleLoadManager and list in bundle admin API

### PR DESCRIPTION
**Pull request by Demogorgon314**
PIP: https://github.com/apache/pulsar/issues/16691

### Motivation
When using `ExtensibleLoadManager` and list in bundle admin API,
it will redirect forever because `isServiceUnitOwned` method is checking the `ownershipCache` as the ownership storage, however, when using `ExtensibleLoadManager`, it stored the ownership to table view.

### Modifications

* Call `isServiceUnitOwnedAsync ` when using `isServiceUnitOwned `.
* Add unit test to cover this case.

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete`

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/namespace/NamespaceService.java [hunk 1]
```diff
@@ -1108,19 +1108,7 @@ public Set<NamespaceBundle> getOwnedServiceUnits() {
     }
 
     public boolean isServiceUnitOwned(ServiceUnitId suName) throws Exception {
```
Comment (eolivelli) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/namespace/NamespaceService.java:1110]: as a follow up work we should remove this synchronous method

**Review by eolivelli — APPROVED** [review]
+1

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/namespace/NamespaceService.java [hunk 2]
```diff
@@ -1108,19 +1108,7 @@ public Set<NamespaceBundle> getOwnedServiceUnits() {
...
-        if (suName instanceof TopicName) {
-            return isTopicOwnedAsync((TopicName) suName).get();
```
Comment (heesung-sn) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/namespace/NamespaceService.java:1112]: Are we safe to skip these checks?

**Review by heesung-sn — APPROVED** [review]

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/namespace/NamespaceService.java (hunk 2 above)
Comment (Demogorgon314) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/namespace/NamespaceService.java:1112]: Yes, you can see the `isServiceUnitOwnedAsync` method has the same checks.

**Review by gaoran10 — APPROVED** [review]

**Review by mattisonchao — APPROVED** [review]

**Review by codelipenghui — APPROVED** [review]

**Pull request merged**